lib/node_modules/@stdlib/ndarray/some-by/lib/assign.js

```diff
@@ -0,0 +1,205 @@
+var maybeBroadcastArray = require( '@stdlib/ndarray/base/maybe-broadcast-array' );
+var broadcastScalar = require( '@stdlib/ndarray/base/broadcast-scalar' );
+var objectAssign = require( '@stdlib/object/assign' );
+var zeroTo = require( '@stdlib/array/base/zero-to' );
+var format = require( '@stdlib/string/format' );
+var defaults = require( './defaults.json' );
+var validate = require( './validate.js' );
+
+
+// VARIABLES //
+
+var TYPES = [
+	'int8',
+	'int16',
+	'int32',
+	'uint8',
+	'uint8c',
+	'uint16',
+	'uint32'
+];
+
+
+// MAIN //
+
+/**
+* Tests whether at least `n` elements in an ndarray pass a test implemented by a predicate function along one or more dimensions.
+*
+* @param {ndarray} x - input ndarray
+* @param {(ndarray|*)} n - number of elements which must pass the test
+* @param {ndarray} y - output ndarray
+* @param {Options} [options] - function options
+* @param {IntegerArray} [options.dims] - list of dimensions over which to perform a reduction
+* @param {Function} predicate - predicate function
+* @param {*} [thisArg] - predicate execution context
+* @throws {TypeError} first argument must be an ndarray-like object
+* @throws {Error} second argument must be broadcast-compatible with the non-reduced dimensions of the input ndarray
+* @throws {TypeError} third argument must be an ndarray-like object
+* @throws {TypeError} options argument must be an object
+* @throws {Error} must provide valid options
+* @throws {TypeError} predicate must be a function
+* @returns {ndarray} output ndarray
+*
+* @example
+* var Float64Array = require( '@stdlib/array/float64' );
+* var ndarray = require( '@stdlib/ndarray/ctor' );
+* var isEven = require( '@stdlib/assert/is-even' ).isPrimitive;
+* var empty = require( '@stdlib/ndarray/empty' );
+*
+* // Create a data buffer:
+* var xbuf = new Float64Array( [ 1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0, 10.0, 11.0, 12.0 ] );
+*
+* // Define the shape of the input array:
+* var shape = [ 3, 1, 2 ];
+*
+* // Define the array strides:
+* var sx = [ 4, 4, 1 ];
+*
+* // Define the index offset:
+* var ox = 1;
+*
+* // Create an input ndarray:
+* var x = new ndarray( 'float64', xbuf, shape, sx, ox, 'row-major' );
+*
+* // Create an output ndarray:
+* var y = empty( [], {
+*     'dtype': 'bool'
+* });
+*
+* // Perform reduction:
+* var out = assign( x, 3, y, isEven );
+* // returns <ndarray>
+*
+* var v = out.get();
+* // returns true
+*/
+function assign( x, n, y, options, predicate, thisArg ) {
+	var nargs;
+	var opts;
+	var err;
+	var ord;
+	var ctx;
+	var cb;
+	var N;
+	var v;
+
+	if ( !isndarrayLike( x ) ) {
+		throw new TypeError( format( 'invalid argument. First argument must be an ndarray-like object. Value: `%s`.', x ) );
+	}
+	if ( !isndarrayLike( y ) ) {
+		throw new TypeError( format( 'invalid argument. Third argument must be an ndarray-like object. Value: `%s`.', y ) );
+	}
+
+	N = ndims( x );
+	nargs = arguments.length;
+
+	if ( nargs === 4 ) {
```

Comment:
Also, as in `ndarray/count-if`, add comments regarding the signatures. This helps you (a) reason about the code and (b) helps future readers.